src/main/java/org/veupathdb/service/eda/compute/plugins/differentialabundance/DifferentialAbundancePlugin.java

```diff
@@ -0,0 +1,130 @@
+    System.out.println(computeEntityIdVarSpec);
+    String computeEntityIdColName = util.toColNameOrEmpty(computeEntityIdVarSpec);
+    System.out.println(computeEntityIdColName);
+    String method = computeConfig.getDifferentialAbundanceMethod().getValue();
+    String groupA = "c('Female')";
+    String groupB = "c('Male')";
+    // String groupA = computeConfig.getDifferentialAbundanceGroupA() != null ? listToRVector(computeConfig.getDifferentialAbundanceGroupA()) : "NULL";
+    // String groupB = computeConfig.getDifferentialAbundanceGroupB() != null ? listToRVector(computeConfig.getDifferentialAbundanceGroupB()) : "NULL";
+
+    // Needs to be in a dot notation... like comptueEntityIdColName prolly
+    VariableSpec comparisonVariableSpec = computeConfig.getComparisonVariable();
+    String comparisonVariable = util.toColNameOrEmpty(comparisonVariableSpec);
+    HashMap<String, InputStream> dataStream = new HashMap<>();
+    dataStream.put(INPUT_DATA, getWorkspace().openStream(INPUT_DATA));
+    List<VariableDef> idColumns = new ArrayList<>();
+    for (EntityDef ancestor : meta.getAncestors(entity)) {
+      idColumns.add(ancestor.getIdColumnDef());
+    }
+    
+    RServe.useRConnectionWithRemoteFiles(dataStream, connection -> {
+      connection.voidEval("print('starting differential abundance computation')");
+
+      // Read in the abundance data
+      List<VariableSpec> computeInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      computeInputVars.addAll(util.getChildrenVariables(computeConfig.getCollectionVariable()));
+      computeInputVars.addAll(idColumns);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, computeInputVars));
+      connection.voidEval("absoluteAbundanceData <- " + INPUT_DATA); // Need to rename so we can go get the sampleMetadata later
+
+      // Read in the sample metadata
+      List<VariableSpec> sampleMetadataVars = ListBuilder.asList(comparisonVariableSpec);
+      sampleMetadataVars.add(computeEntityIdVarSpec);
+      System.out.println(sampleMetadataVars);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, sampleMetadataVars));
+      connection.voidEval("sampleMetadata <- " + INPUT_DATA);
+
+
+      // TODO make a helper for this i think
+      List<String> dotNotatedIdColumns = idColumns.stream().map(VariableDef::toDotNotation).collect(Collectors.toList());
+      String dotNotatedIdColumnsString = "c(";
+      boolean first = true;
+      for (String idCol : dotNotatedIdColumns) {
+        if (first) {
+          first = false;
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + util.singleQuote(idCol);
+        } else {
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + "," + util.singleQuote(idCol);
+        }
+      }
+      dotNotatedIdColumnsString = dotNotatedIdColumnsString + ")";
+
+      // TEMP FOR TESTING ONLY - REMOVE WHEN ABSOLUTE ABUNDANCES ARE HERE
```

Comment:
how is that going btw? getting absolute abundances loaded and collections annotated and constrained?

Reply:
lol bad. I keep poking people and getting radio silence.